src/sandstorm/run-bundle.c++

```diff
@@ -1556,36 +1563,65 @@ private:
     }
   }
 
-  pid_t startMongo(const Config& config) {
+  pid_t startMongo(const Config& config, bool doMongoRepair) {
     Subprocess process([&]() -> int {
       dropPrivs(config.uids);
       clearSignalMask();
 
-      KJ_SYSCALL(execl("/bin/mongod", "/bin/mongod", "--fork",
-          "--bind_ip", "127.0.0.1", "--port", kj::str(config.mongoPort).cStr(),
-          "--dbpath", "/var/mongo", "--logpath", "/var/log/mongo.log",
-          "--pidfilepath", "/var/pid/mongo.pid",
-          "--auth", "--nohttpinterface", "--noprealloc", "--nopreallocj", "--smallfiles",
-          "--replSet", "ssrs", "--oplogSize", "16",
-          EXEC_END_ARGS));
+      kj::Vector<const char*> args;
+      args.add("/bin/mongod");
+      args.add("--fork");
+      args.add("--bind_ip");
+      args.add("127.0.0.1");
+      args.add("--port");
+      args.add(kj::str(config.mongoPort).cStr());
```

Comment:
kj::str here returns a temporary string which, because it's not being stored anywhere, is destroyed when the line finishes. But you've stored a pointer to said string into the vector, so that pointer is now dangling, possibly causing the segfault.